go/private/extensions.bzl

```diff
@@ -1,6 +1,24 @@
-load("//go/private:sdk.bzl", "go_download_sdk_rule", "go_host_sdk_rule", "go_multiple_toolchains")
+load("//go/private:sdk.bzl", "detect_host_platform", "go_download_sdk_rule", "go_host_sdk_rule", "go_multiple_toolchains")
 load("//go/private:repositories.bzl", "go_rules_dependencies")
 
+def host_compatible_toolchain_impl(ctx):
+    ctx.file("BUILD.bazel", content = "")
+    host_compatible_sdk = ctx.attr.toolchains[0]
+    ctx.file("defs.bzl", content = """
+host_compatible_sdk = Label("{}")
+""".format(host_compatible_sdk))
+
+host_compatible_toolchain = repository_rule(
+    implementation = host_compatible_toolchain_impl,
+    attrs = {
+        "toolchains": attr.string_list(
+            doc = "A non empty list of host compatible toolchains",
```

Comment:
We could be more specific here: We pass in the label of the `ROOT` file in such a toolchain's repository.